`atomicapp fetch` errors with traceback if docker daemon is not running

```bash
$ sudo atomicapp fetch --destination localdir projectatomic/helloapache  -v                                                
2016-03-11 18:31:42,293 - [INFO] - cli/main.py - Action/Mode Selected is: fetch
2016-03-11 18:31:42,293 - [DEBUG] - cli/main.py - Final parsed cmdline: fetch --destination localdir projectatomic/helloapache -v
2016-03-11 18:31:42,294 - [DEBUG] - nulecule/main.py - NuleculeManager init app_path: localdir
2016-03-11 18:31:42,294 - [DEBUG] - nulecule/main.py - NuleculeManager init image: projectatomic/helloapache
2016-03-11 18:31:42,294 - [DEBUG] - nulecule/main.py - Request to unpack to projectatomic/helloapache to localdir
2016-03-11 18:31:42,294 - [INFO] - nulecule/base.py - Unpacking image: projectatomic/helloapache to localdir
2016-03-11 18:31:42,343 - [ERROR] - cli/main.py - Could not connect to the docker daemon.
Traceback (most recent call last):
  File "/home/vagrant/atomicapp/atomicapp/cli/main.py", line 77, in cli_fetch
    nm.fetch(**argdict)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/main.py", line 186, in fetch
    self.nulecule = self.unpack(update, dryrun, config=self.answers)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/main.py", line 129, in unpack
    nodeps=nodeps, dryrun=dryrun, update=update)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/base.py", line 103, in unpack
    docker_handler = DockerHandler(dryrun=dryrun)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/container.py", line 42, in __init__
    raise DockerException("Could not connect to the "
DockerException: Could not connect to the docker daemon.
```

--- a/atomicapp/cli/main.py
+++ b/atomicapp/cli/main.py
@@ -51,84 +51,53 @@ def print_app_location(app_path):
 
 
 def cli_genanswers(args):
-    try:
-        argdict = args.__dict__
-        nm = NuleculeManager(app_spec=argdict['app_spec'],
-                             destination='none')
-        nm.genanswers(**argdict)
-        Utils.rm_dir(nm.app_path)  # clean up files
-        sys.exit(0)
-    except NuleculeException as e:
-        logger.error(e)
-        sys.exit(1)
-    except Exception as e:
-        logger.error(e, exc_info=True)
-        sys.exit(1)
+    argdict = args.__dict__
+    nm = NuleculeManager(app_spec=argdict['app_spec'],
+                         destination='none')
+    nm.genanswers(**argdict)
+    Utils.rm_dir(nm.app_path)  # clean up files
+    sys.exit(0)
 
 
 def cli_fetch(args):
-    try:
-        argdict = args.__dict__
-        destination = argdict['destination']
-        nm = NuleculeManager(app_spec=argdict['app_spec'],
-                             destination=destination,
-                             cli_answers=argdict['cli_answers'],
-                             answers_file=argdict['answers'])
-        nm.fetch(**argdict)
-        # Clean up the files if the user asked us to. Otherwise
-        # notify the user where they can manage the application
-        if destination and destination.lower() == 'none':
-            Utils.rm_dir(nm.app_path)
-        else:
-            print_app_location(nm.app_path)
-        sys.exit(0)
-    except NuleculeException as e:
-        logger.error(e)
-        sys.exit(1)
-    except Exception as e:
-        logger.error(e, exc_info=True)
-        sys.exit(1)
+    argdict = args.__dict__
+    destination = argdict['destination']
+    nm = NuleculeManager(app_spec=argdict['app_spec'],
+                         destination=destination,
+                         cli_answers=argdict['cli_answers'],
+                         answers_file=argdict['answers'])
+    nm.fetch(**argdict)
+    # Clean up the files if the user asked us to. Otherwise
+    # notify the user where they can manage the application
+    if destination and destination.lower() == 'none':
+        Utils.rm_dir(nm.app_path)
+    else:
+        print_app_location(nm.app_path)
+    sys.exit(0)
 
 
 def cli_run(args):
-    try:
-        argdict = args.__dict__
-        destination = argdict['destination']
-        nm = NuleculeManager(app_spec=argdict['app_spec'],
-                             destination=destination,
-                             cli_answers=argdict['cli_answers'],
-                             answers_file=argdict['answers'])
-        nm.run(**argdict)
-        # Clean up the files if the user asked us to. Otherwise
-        # notify the user where they can manage the application
-        if destination and destination.lower() == 'none':
-            Utils.rm_dir(nm.app_path)
-        else:
-            print_app_location(nm.app_path)
-        sys.exit(0)
-    except DockerException as e:
-        logger.error(e)
-        sys.exit(1)
-    except NuleculeException as e:
-        logger.error(e)
-        sys.exit(1)
-    except Exception as e:
-        logger.error(e, exc_info=True)
-        sys.exit(1)
+    argdict = args.__dict__
+    destination = argdict['destination']
+    nm = NuleculeManager(app_spec=argdict['app_spec'],
+                         destination=destination,
+                         cli_answers=argdict['cli_answers'],
+                         answers_file=argdict['answers'])
+    nm.run(**argdict)
+    # Clean up the files if the user asked us to. Otherwise
+    # notify the user where they can manage the application
+    if destination and destination.lower() == 'none':
+        Utils.rm_dir(nm.app_path)
+    else:
+        print_app_location(nm.app_path)
+    sys.exit(0)
 
 
 def cli_stop(args):
-    try:
-        argdict = args.__dict__
-        nm = NuleculeManager(app_spec=argdict['app_spec'])
-        nm.stop(**argdict)
-        sys.exit(0)
-    except NuleculeException as e:
-        logger.error(e)
-        sys.exit(1)
-    except Exception as e:
-        logger.error(e, exc_info=True)
-        sys.exit(1)
+    argdict = args.__dict__
+    nm = NuleculeManager(app_spec=argdict['app_spec'])
+    nm.stop(**argdict)
+    sys.exit(0)
 
 
 # Create a custom action parser. Need this because for some args we don't
@@ -142,6 +111,20 @@ class TrueOrFalseAction(argparse.Action):
         else:
             booleanvalue = False
         setattr(namespace, self.dest, booleanvalue)
+
+
+def cli_func_exec(cli_func, cli_func_args):
+    try:
+        cli_func(cli_func_args)
+    except DockerException as e:
+        logger.error(e)
+        sys.exit(1)
+    except NuleculeException as e:
+        logger.error(e)
+        sys.exit(1)
+    except Exception as e:
+        logger.error(e, exc_info=True)
+        sys.exit(1)
 
 
 class CLI():
@@ -457,7 +440,7 @@ class CLI():
         lock = LockFile(os.path.join(Utils.getRoot(), LOCK_FILE))
         try:
             lock.acquire(timeout=-1)
-            args.func(args)
+            cli_func_exec(args.func, args)
         except AttributeError:
             if hasattr(args, 'func'):
                 raise

--- a/atomicapp/nulecule/container.py
+++ b/atomicapp/nulecule/container.py
@@ -31,10 +31,18 @@ class DockerHandler(object):
             except subprocess.CalledProcessError as e:
                 if "client and server don't have same version" in e.output \
                         or "client is newer than server" in e.output:
-                    print("\nThe docker version in this Atomic App differs "
-                          "greatly from the host version.\nPlease use a different "
-                          "Atomic App version for this host.\n")
-                raise e
+                    raise DockerException("\nThe docker version in this "
+                                          "Atomic App differs greatly from "
+                                          "the host version.\nPlease use a "
+                                          "different Atomic App version for "
+                                          "this host.\n")
+                elif "Is your docker daemon up and running" in e.output or \
+                     "Are you trying to connect to a TLS-enabled daemon " \
+                     "without TLS" in e.output:
+                    raise DockerException("Could not connect to the "
+                                          "docker daemon.")
+                else:
+                    raise DockerException(e.output)
 
     def pull(self, image, update=False):
         """

